gdal/ogr/ogrsf_frmts/openfilegdb/ogr_openfilegdb.h

```diff
@@ -213,4 +213,26 @@ class OGROpenFileGDBDataSource final: public OGRDataSource
 
 int OGROpenFileGDBIsComparisonOp(int op);
 
+/************************************************************************/
+/*                   OGROpenFileGDBSingleFeatureLayer                   */
+/************************************************************************/
+
+class OGROpenFileGDBSingleFeatureLayer final: public OGRLayer
```

Comment:
I'm not asking to refactor this per this PR but ``grep -r SingleFeature ogr/ogrsf_frmts/`` confirms my intuition that this is a very common pattern accross drivers and we should rather move that functionality to ogr/ogrsf_frmts/generic. I've traced that in https://github.com/OSGeo/gdal/issues/4310

Reply:
Agreed!